system/Files/File.php

```diff
@@ -70,17 +70,43 @@ public function getSize()
         return $this->size ?? ($this->size = parent::getSize());
     }
 
+    /**
+     * Retrieve the file size by unit, calculated in IEC standards with 1024 as base value.
+     *
+     * @phpstan-param positive-int $precision
+     *
+     * @return false|int|string
+     */
+    public function getSizeByUnitBinary(FileSizeUnit $unit = FileSizeUnit::B, int $precision = 3)
+    {
+        return $this->getSizeByUnitInternal(1024, $unit, $precision);
+    }
+
+    /**
+     * Retrieve the file size by unit, calculated in metric standards with 1000 as base value.
+     *
+     * @phpstan-param positive-int $precision
+     *
+     * @return false|int|string
+     */
+    public function getSizeByUnitMetric(FileSizeUnit $unit = FileSizeUnit::B, int $precision = 3)
+    {
+        return $this->getSizeByUnitInternal(1000, $unit, $precision);
+    }
+
     /**
      * Retrieve the file size by unit.
      *
+     * @deprecated Use getSizeByUnitBinary or getSizeByUnitMetric instead
+     *
      * @return false|int|string
      */
     public function getSizeByUnit(string $unit = 'b')
     {
         return match (strtolower($unit)) {
-            'kb'    => number_format($this->getSize() / 1024, 3),
-            'mb'    => number_format(($this->getSize() / 1024) / 1024, 3),
-            default => $this->getSize(),
+            'kb'    => $this->getSizeByUnitBinary(FileSizeUnit::KB),
+            'mb'    => $this->getSizeByUnitBinary(FileSizeUnit::MB),
+            default => $this->getSizeByUnitBinary(FileSizeUnit::B)
```

Comment:
There is no need for a new method here.
```suggestion
            default => $this->getSize()
```